pkg/controller/infrastructure/actuator.go

```diff
@@ -31,16 +33,25 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/log"
 )
 
+// State represents the last known State of an Infrastructure resource.
+// It is saved after a reconciliation and used during restore operations.
+type State struct {
```

Comment:
Just a nit, but maybe consider to rename this to `InfrastructureState` to make its purpose more clear.